[Bug Skip][Regression]: UT nightly failures 2026-6-24

### 🐛 Describe the bug with skip template

Cases:
~~op_ut,third_party.torch-xpu-ops.test.xpu.export.test_export_xpu.TestExport,test_module~~
~~op_ut,third_party.torch-xpu-ops.test.xpu.higher_order_ops.test_invoke_subgraph_xpu.TestInvokeSubgraphCompile,test_triton_kernel_native~~
op_ut,third_party.torch-xpu-ops.test.xpu.profiler.test_profiler_xpu.TestExperimentalUtils,test_utils_get_optimizable_events
op_ut,third_party.torch-xpu-ops.test.xpu.profiler.test_profiler_xpu.TestExperimentalUtils,test_utils_intervals_overlap
op_ut,third_party.torch-xpu-ops.test.xpu.profiler.test_profiler_xpu.TestExperimentalUtils,test_utils_compute_queue_depth
op_ut,third_party.torch-xpu-ops.test.xpu.profiler.test_profiler_xpu.TestExperimentalUtils,test_utils_compute_self_time
op_ut,third_party.torch-xpu-ops.test.xpu.profiler.test_profiler_xpu.TestExperimentalUtils,test_utils_compute_queue_depth_when_no_cuda_events
op_ut,third_party.torch-xpu-ops.test.xpu.profiler.test_profiler_xpu.TestExperimentalUtils,test_utils_compute_idle_time
~~op_ut,third_party.torch-xpu-ops.test.xpu.test_custom_ops_xpu.TestCustomOpAPI,test_register_autograd_defaults~~
~~op_ut,third_party.torch-xpu-ops.test.xpu.test_custom_ops_xpu.TestCustomOpAPI,test_register_autograd_error_cases~~
~~op_ut,third_party.torch-xpu-ops.test.xpu.test_custom_ops_xpu.TestCustomOpAPI,test_register_autograd_kwargonly_low_level~~

## ErrorLog
### AssertionError: False is not true
```bash
____________________________ TestExport.test_module ____________________________
[gw4] linux -- Python 3.10.20 /tmp/xpu-tool/Python/3.10.20/x64/bin/python
Traceback (most recent call last):
  File "/__w/torch-xpu-ops/torch-xpu-ops/pytorch/third_party/torch-xpu-ops/test/xpu/export/test_export_xpu.py", line 9005, in test_module
    self.assertTrue(
  File "/tmp/xpu-tool/Python/3.10.20/x64/lib/python3.10/unittest/case.py", line 687, in assertTrue
    raise self.failureException(msg)
AssertionError: False is not true
```
### AssertionError: 'clas[1125 chars]': o});  x = y = triton_kernel_wrapper_mutatio[83 chars],)\n' != 'clas[1125 chars]': o}, launch_kwargs = (\'BLOCK_SIZE\',));  x [118 chars],)\n'
```bash
_____________ TestInvokeSubgraphCompile.test_triton_kernel_native ______________
[gw5] linux -- Python 3.10.20 /tmp/xpu-tool/Python/3.10.20/x64/bin/python
Traceback (most recent call last):
  File "/__w/torch-xpu-ops/torch-xpu-ops/pytorch/third_party/torch-xpu-ops/test/xpu/higher_order_ops/test_invoke_subgraph_xpu.py", line 2498, in test_triton_kernel_native
    self.assertExpectedInline(
  File "/tmp/xpu-tool/Python/3.10.20/x64/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 3534, in assertExpectedInline
    return super().assertExpectedInline(actual if isinstance(actual, str) else str(actual), expect, skip + 1)
  File "/tmp/xpu-tool/Python/3.10.20/x64/lib/python3.10/site-packages/expecttest/__init__.py", line 413, in assertExpectedInline
    assert_expected_inline(
  File "/tmp/xpu-tool/Python/3.10.20/x64/lib/python3.10/site-packages/expecttest/__init__.py", line 378, in assert_expected_inline
    assert_eq(expect, actual, msg=help_text)
  File "/tmp/xpu-tool/Python/3.10.20/x64/lib/python3.10/site-packages/expecttest/__init__.py", line 450, in assertMultiLineEqualMaybeCppStack
    self.assertMultiLineEqual(expect, actual, *args, **kwargs)
  File "/tmp/xpu-tool/Python/3.10.20/x64/lib/python3.10/unittest/case.py", line 1226, in assertMultiLineEqual
    self.fail(self._formatMessage(msg, standardMsg))
  File "/tmp/xpu-tool/Python/3.10.20/x64/lib/python3.10/unittest/case.py", line 675, in fail
    raise self.failureException(msg)
AssertionError: 'clas[1125 chars]': o});  x = y = triton_kernel_wrapper_mutatio[83 chars],)\n' != 'clas[1125 chars]': o}, launch_kwargs = (\'BLOCK_SIZE\',));  x [118 chars],)\n'
  class GraphModule(torch.nn.Module):
      def forward(self, L_x_: "f32[5]", L_y_: "f32[5]"):
          l_x_ = L_x_
          l_y_ = L_y_
  
          x: "f32[5]" = l_x_.sin();  l_x_ = None
  
          y: "f32[5]" = l_y_.sin();  l_y_ = None
  
          subgraph_0 = self.subgraph_0
          invoke_subgraph = torch.ops.higher_order.invoke_subgraph(subgraph_0, 'subgraph_0', x, y);  subgraph_0 = x = None
          z: "f32[5]" = invoke_subgraph[0];  invoke_subgraph = None
  
          subgraph_1 = self.subgraph_0
          invoke_subgraph_1 = torch.ops.higher_order.invoke_subgraph(subgraph_1, 'subgraph_0', z, y);  subgraph_1 = z = y = None
          getitem_1: "f32[5]" = invoke_subgraph_1[0];  invoke_subgraph_1 = None
          return (getitem_1,)
  
      class subgraph_0(torch.nn.Module):
          def forward(self, x: "f32[5]", y: "f32[5]"):
              o: "f32[5]" = torch.zeros_like(x)
  
-             triton_kernel_wrapper_mutation = torch.ops.higher_order.triton_kernel_wrapper_mutation(kernel_idx = 0, constant_args_idx = 0, grid = [(5, 1, 1)], tma_descriptor_metadata = {}, kwargs = {'in_ptr0': x, 'in_ptr1': y, 'out_ptr': o});  x = y = triton_kernel_wrapper_mutation = None
+             triton_kernel_wrapper_mutation = torch.ops.higher_order.triton_kernel_wrapper_mutation(kernel_idx = 0, constant_args_idx = 0, grid = [(5, 1, 1)], tma_descriptor_metadata = {}, kwargs = {'in_ptr0': x, 'in_ptr1': y, 'out_ptr': o}, launch_kwargs = ('BLOCK_SIZE',));  x = y = triton_kernel_wrapper_mutation = None
?                                                                                                                                                                                                                                                +++++++++++++++++++++++++++++++++
  
              sin: "f32[5]" = o.sin();  o = None
              return (sin,)
 : To accept the new output, re-run test with envvar EXPECTTEST_ACCEPT=1 (we recommend staging/committing your changes before doing this)
```
### AttributeError: module 'torch.profiler._utils' has no attribute 'BasicEvaluation'
```bash
___________ TestExperimentalUtils.test_utils_get_optimizable_events ____________
[gw7] linux -- Python 3.10.20 /tmp/xpu-tool/Python/3.10.20/x64/bin/python
Traceback (most recent call last):
  File "/__w/torch-xpu-ops/torch-xpu-ops/pytorch/third_party/torch-xpu-ops/test/xpu/profiler/test_profiler_xpu.py", line 3292, in test_utils_get_optimizable_events
    basic_evaluation = _utils.BasicEvaluation(self.load_mock_profile())
AttributeError: module 'torch.profiler._utils' has no attribute 'BasicEvaluation'

_____________ TestExperimentalUtils.test_utils_compute_queue_depth _____________
[gw4] linux -- Python 3.10.20 /tmp/xpu-tool/Python/3.10.20/x64/bin/python
Traceback (most recent call last):
  File "/__w/torch-xpu-ops/torch-xpu-ops/pytorch/third_party/torch-xpu-ops/test/xpu/profiler/test_profiler_xpu.py", line 3214, in test_utils_compute_queue_depth
    basic_evaluation = _utils.BasicEvaluation(profiler)
AttributeError: module 'torch.profiler._utils' has no attribute 'BasicEvaluation

______________ TestExperimentalUtils.test_utils_compute_self_time ______________
[gw7] linux -- Python 3.10.20 /tmp/xpu-tool/Python/3.10.20/x64/bin/python
Traceback (most recent call last):
  File "/__w/torch-xpu-ops/torch-xpu-ops/pytorch/third_party/torch-xpu-ops/test/xpu/profiler/test_profiler_xpu.py", line 3182, in test_utils_compute_self_time
    basic_eval = _utils.BasicEvaluation(prof.profiler)
AttributeError: module 'torch.profiler._utils' has no attribute 'BasicEvaluation'

___ TestExperimentalUtils.test_utils_compute_queue_depth_when_no_cuda_events ___
[gw5] linux -- Python 3.10.20 /tmp/xpu-tool/Python/3.10.20/x64/bin/python
Traceback (most recent call last):
  File "/__w/torch-xpu-ops/torch-xpu-ops/pytorch/third_party/torch-xpu-ops/test/xpu/profiler/test_profiler_xpu.py", line 3261, in test_utils_compute_queue_depth_when_no_cuda_events
    basic_evaluation = _utils.BasicEvaluation(prof.profiler)
AttributeError: module 'torch.profiler._utils' has no attribute 'BasicEvaluation'

______________ TestExperimentalUtils.test_utils_compute_idle_time ______________
[gw3] linux -- Python 3.10.20 /tmp/xpu-tool/Python/3.10.20/x64/bin/python
Traceback (most recent call last):
  File "/__w/torch-xpu-ops/torch-xpu-ops/pytorch/third_party/torch-xpu-ops/test/xpu/profiler/test_profiler_xpu.py", line 3266, in test_utils_compute_idle_time
    basic_evaluation = _utils.BasicEvaluation(profiler)
AttributeError: module 'torch.profiler._utils' has no attribute 'BasicEvaluation'
```
### AttributeError: module 'torch.profiler._utils' has no attribute 'EventKey'
```bash
______________ TestExperimentalUtils.test_utils_intervals_overlap ______________
[gw7] linux -- Python 3.10.20 /tmp/xpu-tool/Python/3.10.20/x64/bin/python
Traceback (most recent call last):
  File "/__w/torch-xpu-ops/torch-xpu-ops/pytorch/third_party/torch-xpu-ops/test/xpu/profiler/test_profiler_xpu.py", line 3193, in test_utils_intervals_overlap
    event = _utils.EventKey(MockProfilerEvent("Event 1", 1, 5, 5))
AttributeError: module 'torch.profiler._utils' has no attribute 'EventKey'
```
### RuntimeError: element 0 of tensors does not require grad and does not have a grad_fn
```bash
______________ TestCustomOpAPI.test_register_autograd_defaults ________________
[gw7] linux -- Python 3.10.20 /tmp/xpu-tool/Python/3.10.20/x64/bin/python
Traceback (most recent call last):
  File "/__w/torch-xpu-ops/torch-xpu-ops/pytorch/third_party/torch-xpu-ops/test/xpu/test_custom_ops_xpu.py", line 2874, in test_register_autograd_defaults
    torch.ops._torch_testing.foo(w, z=42).sum().backward()
  File "/tmp/xpu-tool/Python/3.10.20/x64/lib/python3.10/site-packages/torch/_tensor.py", line 623, in backward
    torch.autograd.backward(
  File "/tmp/xpu-tool/Python/3.10.20/x64/lib/python3.10/site-packages/torch/autograd/__init__.py", line 395, in backward
    _engine_run_backward(
  File "/tmp/xpu-tool/Python/3.10.20/x64/lib/python3.10/site-packages/torch/autograd/graph.py", line 979, in _engine_run_backward
    return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
RuntimeError: element 0 of tensors does not require grad and does not have a grad_fn

______________ TestCustomOpAPI.test_register_autograd_error_cases ______________
[gw7] linux -- Python 3.10.20 /tmp/xpu-tool/Python/3.10.20/x64/bin/python
RuntimeError: element 0 of tensors does not require grad and does not have a grad_fn

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/__w/torch-xpu-ops/torch-xpu-ops/pytorch/third_party/torch-xpu-ops/test/xpu/test_custom_ops_xpu.py", line 3828, in test_register_autograd_error_cases
    with self.assertRaisesRegex(RuntimeError, "no autograd formula"):
  File "/tmp/xpu-tool/Python/3.10.20/x64/lib/python3.10/unittest/case.py", line 239, in __exit__
    self._raiseFailure('"{}" does not match "{}"'.format(
  File "/tmp/xpu-tool/Python/3.10.20/x64/lib/python3.10/unittest/case.py", line 163, in _raiseFailure
    raise self.test_case.failureException(msg)
AssertionError: "no autograd formula" does not match "element 0 of tensors does not require grad and does not have a grad_fn"

__________ TestCustomOpAPI.test_register_autograd_kwargonly_low_level __________
[gw7] linux -- Python 3.10.20 /tmp/xpu-tool/Python/3.10.20/x64/bin/python
Traceback (most recent call last):
  File "/__w/torch-xpu-ops/torch-xpu-ops/pytorch/third_party/torch-xpu-ops/test/xpu/test_custom_ops_xpu.py", line 2835, in test_register_autograd_kwargonly_low_level
    torch.ops._torch_testing.foo(x, y=3.14).sum().backward()
  File "/tmp/xpu-tool/Python/3.10.20/x64/lib/python3.10/site-packages/torch/_tensor.py", line 623, in backward
    torch.autograd.backward(
  File "/tmp/xpu-tool/Python/3.10.20/x64/lib/python3.10/site-packages/torch/autograd/__init__.py", line 395, in backward
    _engine_run_backward(
  File "/tmp/xpu-tool/Python/3.10.20/x64/lib/python3.10/site-packages/torch/autograd/graph.py", line 979, in _engine_run_backward
    return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
RuntimeError: element 0 of tensors does not require grad and does not have a grad_fn
```
## Pytorch Version
latest good : 12b766bf533ff188b4d861184c54b7278ceaca9a 
current : 206283a284889f3a4f62510f8b5a8068d41121c4 

### Versions

<details>
<summary>Detail</summary>
Collecting environment information...
PyTorch version: 2.14.0a0+git206283a
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.4 LTS (x86_64)
GCC version: (Ubuntu 13.3.0-6ubuntu2~24.04.1) 13.3.0
Clang version: Could not collect
CMake version: version 3.31.6
Libc version: glibc-2.39

Python version: 3.10.20 (main, Jun 16 2026, 21:01:32) [GCC 13.3.0] (64-bit runtime)
Python platform: Linux-5.15.0-179-generic-x86_64-with-glibc2.39
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
Is XPU available: True
XPU used to build PyTorch: 20260000
Intel GPU driver version:
* intel-opencl-icd:	25.18.33578.51-1146~24.04
* libze1:	1.24.0.0-1146~24.04
Intel GPU models onboard:
* Intel(R) Data Center GPU Max 1100
* Intel(R) Data Center GPU Max 1100
* Intel(R) Data Center GPU Max 1100
* Intel(R) Data Center GPU Max 1100
* Intel(R) Data Center GPU Max 1100
* Intel(R) Data Center GPU Max 1100
* Intel(R) Data Center GPU Max 1100
Vulnerability Mmio stale data:           Not affected
Vulnerability Reg file data sampling:    Not affected
Vulnerability Retbleed:                  Not affected
Vulnerability Spec rstack overflow:      Not affected
Vulnerability Spec store bypass:         Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:                Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:                Mitigation; Enhanced / Automatic IBRS; IBPB conditional; PBRSB-eIBRS SW sequence; BHI BHI_DIS_S
Vulnerability Srbds:                     Not affected
Vulnerability Tsa:                       Not affected
Vulnerability Tsx async abort:           Not affected
Vulnerability Vmscape:                   Mitigation; IBPB before exit to userspace

Versions of relevant libraries:
[pip3] dpcpp-cpp-rt==2026.0.0
[pip3] impi-rt==2021.18.0
[pip3] intel-cmplr-lib-rt==2026.0.0
[pip3] intel-cmplr-lib-ur==2026.0.0
[pip3] intel-cmplr-lic-rt==2026.0.0
[pip3] intel-opencl-rt==2026.0.0
[pip3] intel-openmp==2026.0.0
[pip3] intel-pti==0.17.0
[pip3] intel-sycl-rt==2026.0.0
[pip3] mkl==2026.0.0
[pip3] mypy==1.16.0
[pip3] mypy_extensions==1.1.0
[pip3] numpy==1.24.4
[pip3] nvidia-cuda-cupti==13.3.35
[pip3] oneccl==2022.0.0
[pip3] oneccl-devel==2022.0.0
[pip3] onemkl-license==2026.0.0
[pip3] onemkl-sycl-blas==2026.0.0
[pip3] onemkl-sycl-dft==2026.0.0
[pip3] onemkl-sycl-lapack==2026.0.0
[pip3] onemkl-sycl-rng==2026.0.0
[pip3] onemkl-sycl-sparse==2026.0.0
[pip3] onnx==1.21.0
[pip3] onnx-ir==0.1.16
[pip3] onnxscript==0.6.2
[pip3] optree==0.13.0
[pip3] tbb==2023.0.0
[pip3] tcmlib==1.5.0
[pip3] torch==2.14.0a0+git206283a
[pip3] torchao==0.18.0.dev20260624+xpu
[pip3] torchaudio==2.11.0a0+c0cbdb9
[pip3] torchvision==0.28.0a0+85a1dcc
[pip3] triton-xpu==3.7.2+git5fcc14d9
[pip3] umf==1.1.0
[conda] No relevant packages
</details>

--- a/test/xpu/higher_order_ops/test_invoke_subgraph_xpu.py
+++ b/test/xpu/higher_order_ops/test_invoke_subgraph_xpu.py
@@ -2454,10 +2454,7 @@ class GraphModule(torch.nn.Module):
             if grid_type == 0:
                 grid = (x.numel(),)
             elif grid_type == 1:
-
-                def grid(meta):
-                    return (triton.cdiv(n_elements, meta["BLOCK_SIZE"]),)
-
+                grid = lambda meta: (triton.cdiv(n_elements, meta["BLOCK_SIZE"]),)
             else:
                 grid = grid_fn
 
@@ -2520,7 +2517,7 @@ class GraphModule(torch.nn.Module):
         def forward(self, x: "f32[5]", y: "f32[5]"):
             o: "f32[5]" = torch.zeros_like(x)
 
-            triton_kernel_wrapper_mutation = torch.ops.higher_order.triton_kernel_wrapper_mutation(kernel_idx = 0, constant_args_idx = 0, grid = [(5, 1, 1)], tma_descriptor_metadata = {}, kwargs = {'in_ptr0': x, 'in_ptr1': y, 'out_ptr': o});  x = y = triton_kernel_wrapper_mutation = None
+            triton_kernel_wrapper_mutation = torch.ops.higher_order.triton_kernel_wrapper_mutation(kernel_idx = 0, constant_args_idx = 0, grid = [(5, 1, 1)], tma_descriptor_metadata = {}, kwargs = {'in_ptr0': x, 'in_ptr1': y, 'out_ptr': o}, launch_kwargs = ('BLOCK_SIZE',));  x = y = triton_kernel_wrapper_mutation = None
 
             sin: "f32[5]" = o.sin();  o = None
             return (sin,)

